worlds/mmbn3/lz10.py

```diff
@@ -0,0 +1,131 @@
+import struct
+
+
```

Comment:
I used ndspy as a reference, but since that was developed for GB, GBA, and DS, not knowing which game it uses, I was able to cut out a lot of the unnecessary checks and alternate cases to use just the GBA algorithm. If it's still too close then I can try to rewrite it, but since it's based off of the algorithm used by the GBA itself it's always going to end up functionally identical.

Reply:
Rewritten using an MIT compatible LZ10 library, https://github.com/magical/nlzss

Since it only operates on files instead of byte streams, I had to do a bit of tweaking from the source code in order to get it to work with this use case, but I got the benefit of cutting out all the LZS11 stuff as well so it's smaller than just including that library would have been in the first place.